lib/benchee/conversion/memory.ex

```diff
@@ -46,6 +46,43 @@ defmodule Benchee.Conversion.Memory do
               }
   }
 
+  # Unit conversion functions
+
+  @type unit :: Unit.t
+  @type unit_atom :: atom
+  @type value_with_unit :: {number, unit}
+  @type value_with_unit_atom :: {number, unit_atom}
+
+  @doc """
+  Converts a value for a specified unit and converts it to the equivalent of another unit of measure.
+
+  ## Examples
+
+    iex> {value, unit} = Benchee.Conversion.Memory.convert({1024, :kilobyte}, :megabyte)
+    iex> value
+    1.0
+    iex> unit.name
+    :megabyte
+  """
+  @spec convert(value_with_unit_atom, unit_atom) :: value_with_unit
+  def convert({value, current_unit_atom}, desired_unit_atom) when is_atom(current_unit_atom) and is_atom(desired_unit_atom) do
+    current_value_unit = {value, unit_for(current_unit_atom)}
+    desired_unit = unit_for(desired_unit_atom)
+    convert(current_value_unit, desired_unit)
+  end
+
+  @spec convert(value_with_unit, unit) :: value_with_unit
```

Comment:
this is strange though, 2 typespecs for convert with the same number of arguments? Is that legal? I think not... @devonestes do you know?

Reply:
Yeah, now that you mention it we really should be defining a single spec for that function. It's funny that Dialyzer wasn't failing in CI, though...

Reply:
If we are only allowed a single spec per function + arity is there another way to convey all of the intended cases of argument lists? As in, we have one for when a ```%Unit``` struct is passed another for handling simply the atom of the name of the unit.

Reply:
You can use a pipe in a typespec, which basically stands for "or this other thing". So, the spec we might want here would be the following:

```elixir
@spec convert(value_with_unit | value_with_unit_atom, Unit.t() | unit_atom) :: value_with_unit
```

The downside is you can't see that all possible combinations of those things don't work, which is a slight bummer. Only two of four possible combinations are really valid in real use, but that's just one of the limitations of Dialyzer and typespecs at the moment.